### ISSUE TITLE
Shape-changing augmentations with a mixed batch_prob transform every row, desynchronize labels, and cannot be inverted

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4497]
A shape-changing 2D augmentation (`RandomCrop`, `Resize`) given a mixed `batch_prob` returns the transformed image for every row, the rows whose gate is 0 included. Its `inverse` then raises or silently skips parts of the undo. This came out of the #4485 review. #4485 fixes #4473, the skipped-crop label shift, and these related defects were left out of it on purpose.

Natural sampling cannot produce this. These classes gate the whole batch (`p_batch = p`; see #4425), so it only appears when `params` are passed by hand, as in replay or a hand-edited `batch_prob`. It is still a public input path, and nothing rejects it.

## Repro (main `dc0af1a4`, torch 2.14.0, CPU)

```python
import torch, kornia.augmentation as K

H, W = 24, 32
img = torch.zeros(4, 1, H, W)
for b, (x, y) in enumerate([(12, 10), (15, 13), (18, 11), (9, 14)]):
    img[b, 0, y, x] = 1.0

for name, make in (("RandomCrop", lambda: K.RandomCrop((20, 26), padding=(1, 2), p=0.5)),
                   ("Resize", lambda: K.Resize((20, 26), p=0.5))):
    aug = make(); torch.manual_seed(3)
    params = aug.forward_parameters((4, 1, H, W))
    params["batch_prob"] = torch.tensor([0.0, 1.0, 0.0, 1.0])
    out = aug(img, params=params)
    print(name, tuple(out.shape))          # (4, 1, 20, 26): rows 0 and 2 are transformed too
    try:
        aug.inverse(out)
    except Exception as e:
        print(name, type(e).__name__, str(e)[:100])
```

## Observed

```
RandomCrop (4, 1, 20, 26)
RandomCrop NotImplementedError `inverse` is only applicable for resample cropping mode. Got slice.
Resize (4, 1, 20, 26)
Resize RuntimeError shape mismatch: value tensor of shape [2, 1, 24, 32] cannot be broadcast to indexing result of shape [2, 1, 20, 26]
```

1. **Image:** with mixed gates and a changed output shape, the per-row blend cannot keep the skipped rows' original size, so its fallback returns the transformed output for all rows. The skipped rows' images change even though their gate is 0.
2. **Labels do not follow those pixels.** With `RandomCrop((20, 26), padding=(1, 2), resample="nearest")` and gate `[0, 1, 0, 1]`, a skipped row's keypoint and its landmark pixel disagree:
   - `slice` mode on main: row 2's pixel is at `(19, 7)` (fully cropped) but its keypoint at `(19, 13)` (padded only).
   - With #4485 applied, every skipped row gets its input keypoint back, e.g. `(12, 10)` against a pixel at `(13, 12)`, in both `slice` and `resample` modes.
   - Applied rows are consistent everywhere.
3. **Inverse:**
   - `Resize.inverse` on the mixed batch raises the shape mismatch above (`GeometricAugmentationBase2D.inverse_inputs` writes 24x32 rows into a 20x26 index).
   - `RandomCrop` in `resample` mode fails the same way.
   - `RandomCrop.inverse_keypoints` / `inverse_boxes` skip `unpad` for the applied rows whenever `not params["batch_prob"].all()` (`kornia/augmentation/_2d/geometric/crop.py:315` and `:333`; the image path has the same guard at `:296`).

## Expected

One contract for a mixed gate on a shape-changing augmentation, applied the same way to images, masks, keypoints, boxes and `inverse`. Either reject a mixed `batch_prob` on these classes with a clear `ValueError`, which is consistent with their whole-batch gate, or define what a skipped row returns and make every data key and the inverse follow it.

## Environment

kornia `dc0af1a4` (main), torch 2.14.0, Python 3.11, macOS arm64, CPU. Related: #4473 (skipped-crop label shift, fixed by #4485), #4425 (`p_batch` exposure).

Posted on behalf of @ducha-aiki by Claude (Opus 5).
